cmd/podman/machine/platform.go

```diff
@@ -4,10 +4,28 @@
 package machine
 
 import (
+	"os"
+	"strings"
+
 	"github.com/containers/podman/v4/pkg/machine"
 	"github.com/containers/podman/v4/pkg/machine/qemu"
+	"github.com/sirupsen/logrus"
 )
 
-func GetSystemDefaultProvider() machine.VirtProvider {
-	return qemu.GetVirtualizationProvider()
+func GetSystemProvider(name string) machine.VirtProvider {
+	provider := name
+	providerOverride, found := os.LookupEnv("CONTAINERS_MACHINE_PROVIDER")
+	if found {
+		provider = providerOverride
+	}
+	provider = strings.ToUpper(strings.TrimSpace(provider))
+	switch provider {
+	case "QEMU":
+		return qemu.GetVirtualizationProvider()
```

Comment:
shouldn't this also accept an empty string because this is the default in containers.conf?
at least you didn't set a different value in containers.conf thus causing the warning on every single command
https://github.com/containers/common/commit/ead2e7dfeb07123d16be0bf355938a3c6f4eafba

Reply:
This is covered in the default section with this
```
	if len(provider) > 0 {
			logrus.Warnf("Unuspported provider specified %q. Will use default provider.", provider)
		}
```

If the string is empty, then no warning will appear. After refactoring to hard errors, the empty value should be explicitly added to allowed to preserve backward compatibility with old configs. You are right.

Reply:
This code was reworked. Now there are separate errors for input, which can't be parsed and for valid input but unsupported on the current platform. No fallback.